http.js

```diff
@@ -393,11 +392,11 @@ exports.read = function (request, qualifier) {
             if (Buffer.isBuffer(data)) {
                 contentType = response.headers['content-type'] || '',
                 match = contentType.match(/charset=([a-z0-9-]+)/i),
-                charset = match ? match[1] : 'utf8';
+                charset = match && iconv.encodingExists(match[1]) ? match[1] : 'utf8';
```

Comment:
Since `read` is supposed to return a string, `text/*` is somewhat implied. However, there is `application/json`, `application/xml` or `application/javascript` where I as a user would expect a string result. XML explicitly uses UTF-8 as the default, JSON and JS both support UTF-8 encoding. So `text*/` is not the only content type to consider for conversion to string.

Is `request.charset` is a good point though. You do use a charset parameter in the `Reader` implementation. Maybe this behavior should be collapsed? Something like this:

 - assume `charset: undefined` as the default for `request` - will return string or buffer, exactly as before
 - as soon as `charset` is user-defined, use that value, exactly as before
 - introduce `charset: 'auto'` - will decode to string according to the Content-Type, favoring UTF-8 in the absence of a header value (or throws, if you prefer, though that would not define behavior in case of a missing `charset` parameter)
 - assume `charset: 'auto'` as the default for `read` - will return string in any case, just as the docs state

That way `request` could benefit from proper automatic string conversion as well, instead of this being a quirk of `read`.